### PR TITLE
Added docs for running containers with different ports

### LINKED ISSUE #2377: How to run horovod in multiple servers with same IP and  with different ports
Environment:

Framework: (TensorFlow, Keras, PyTorch, MXNet)
TensorFlow
Framework version:
2.3
Horovod version:
0.20
MPI version:
mpiexec (OpenRTE) 4.0.0
CUDA version:
10.2
NCCL version:
2.7.8
Python version:
3.7.5
OS and version:

Checklist:

Did you search issues to find if somebody asked this question before?
If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?

Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

I have two server machines A and B on the same IP address, but different ports. I pull the docker file from horovod/horovod and run it. I am successfully able to run the 
`horovodrun -np 2 -H localhost:2 python tensorflow2_keras_mnist.py `successfully on each of servers individually. However when I tried  to  modify the following command  `horovodrun -np 4 -H host1:2,host2:2 -p 12345 python keras_mnist_advanced.py` to 
`horovodrun -np 4 -H 'host1 -p 22':2,'host2 -p 9922':2 -p 12345 python tensorflow2_keras_mnist.py`, it does not seem to recognize the ports specified in the command.

Below is the error that i am coming across:--

Traceback (most recent call last):
 ```
 File "/usr/local/bin/horovodrun", line 8, in <module>
    sys.exit(run_commandline())
  File "/usr/local/lib/python3.7/dist-packages/horovod/runner/launch.py", line 723, in run_commandline
    _run(args)
  File "/usr/local/lib/python3.7/dist-packages/horovod/runner/launch.py", line 713, in _run
    return _run_static(args)
  File "/usr/local/lib/python3.7/dist-packages/horovod/runner/launch.py", line 529, in _run_static
    all_host_names, _ = hosts.parse_hosts_and_slots(args.hosts)
  File "/usr/local/lib/python3.7/dist-packages/horovod/runner/common/util/hosts.py", line 79, in parse_hosts_and_slots
    raise ValueError('Invalid host input, please make sure it has '
ValueError: Invalid host input, please make sure it has format as : worker-0:2,worker-1:2.
```

### PR DESCRIPTION
**Pull request by tgaddair**
Fixes #2377.

### CODE REVIEW

**Review by Qiuzhuang — APPROVED** [review]
This looks good to me, thanks.

File: docs/docker.rst [hunk 1]
```diff
@@ -99,4 +99,27 @@ and enable the IPC_LOCK capability for memory registration:
 You need to specify these additional configuration options on primary and secondary workers.
 
 
+Running containers with different ports
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``horovodrun`` uses SSH to launch processes on the workers. When using Docker, it may be the case that different containers
+are placed on the same host with different ports.  In such cases, you can configure your
```
Comment (sblotner) [inline docs/docker.rst:105]: To make the steps prominent for the reader, after "different ports" you could say, "To run containers with different ports:" and then have the following steps as an ordered list:

1. Configure your ~/.ssh/config file to assign custom host names and ports for each container
   (example)

2. Use ``hordovodrun`` directly as though each container were a separate host with its own IP
   (example)

Comment (tgaddair) [reply docs/docker.rst]: Good suggestion!  Updated.

**Review by sblotner — APPROVED** [review]
Steps are very clear!

**Pull request merged**